src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -311,14 +311,24 @@ private CookieOptions BuildCookieOptions()
             Response.Headers[HeaderNames.CacheControl] = HeaderValueNoCache;
             Response.Headers[HeaderNames.Pragma] = HeaderValueNoCache;
             Response.Headers[HeaderNames.Expires] = HeaderValueMinusOne;
+
             if (shouldRedirectToReturnUrl && Response.StatusCode == 200)
             {
+                CookieRedirectContext redirectContext = null;
+                
                 var query = Request.Query;
                 var redirectUri = query[Options.ReturnUrlParameter];
```

Comment:
Original code, didn't think twice. But apparently you suggested a better version. I'll updater.